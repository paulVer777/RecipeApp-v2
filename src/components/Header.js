import React from 'react'
import {Link} from 'react-router-dom'

const Header = () => (
    <div className = 'header'>
        <Link to = {'/'} >Home</Link>
        <Link to = {'/my-ingredients'} >My Ingredients</Link>
    </div>
)

export default Header