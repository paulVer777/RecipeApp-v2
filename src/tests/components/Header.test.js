import ReactShallowRenderer from 'react-test-renderer/shallow'
import React from 'react'
import Header from '../../components/Header';
import {shallow} from 'enzyme'

test('Should render header correctly' , () => {

    const renderer = new ReactShallowRenderer()
    renderer.render(<Header />)
    expect(renderer.getRenderOutput())
})

//After configuration jest with enzyme 

test('should render header correctly', () => {

    const wrapper = shallow(<Header />) // now we can acces to full api https://airbnb.io/enzyme/docs/api/shallow.html (various methods)
    expect(wrapper).toMatchSnapshot()
})


