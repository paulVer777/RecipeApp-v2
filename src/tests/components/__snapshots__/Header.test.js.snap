// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div
  className="header"
>
  <Link
    replace={false}
    to="/"
  >
    Home
  </Link>
  <Link
    replace={false}
    to="/my-ingredients"
  >
    My Ingredients
  </Link>
</div>
`;
